### PR TITLE
Fix Clang-15 warnings

### PR DESCRIPTION
**Pull request by AtariDreams**
Mostly dealing with initialization and codestyle.

### CODE REVIEW

File: cppwinrt/component_writers.h [hunk 1]
```diff
@@ -289,7 +289,7 @@ catch (...) { return winrt::to_hresult(); }
             bind<write_consume_args>(signature));
     }
 
-    void write_component_static_forwarder(writer& w, MethodDef const& method)
+    inline void write_component_static_forwarder(writer& w, MethodDef const& method)
```
Comment (kennykerr) [inline cppwinrt/component_writers.h:292]:
```suggestion
    static void write_component_static_forwarder(writer& w, MethodDef const& method)
```

File: cppwinrt/component_writers.h [hunk 2]
```diff
@@ -643,8 +643,8 @@ catch (...) { return winrt::to_hresult(); }
         {
             if (!info.base && info.is_default)
             {
-                auto methods = info.type.MethodList();
-                offset += methods.second - methods.first;
+                auto [fst, snd] = info.type.MethodList();
+                offset += snd - fst;
```
Comment (kennykerr) [inline cppwinrt/component_writers.h:647]:
```suggestion
                auto [first, second] = info.type.MethodList();
                offset += second - first;
```

File: cppwinrt/cmd_reader.h [hunk 3]
```diff
@@ -195,11 +195,10 @@ namespace cppwinrt
     inline std::string get_module_path()
     {
         std::string path(100, '?');
-        DWORD actual_size{};
 
-        while (true)
+        for (;;)
```
Comment (sylveon) [inline cppwinrt/cmd_reader.h:199]: I don't see how `for (;;)` is any clearer/better than `while (true)`

File: cppwinrt/text_writer.h [hunk 4]
```diff
@@ -341,7 +341,7 @@ namespace cppwinrt
             std::string_view::size_type current_pos{ 0 };
             auto on_new_line = writer_base<T>::back() == '\n';
 
-            while (true)
+            for (;;)
```
Comment (sylveon) [inline cppwinrt/text_writer.h:344]: Idem

File: natvis/cppwinrt_visualizer.cpp [hunk 5]
```diff
@@ -93,7 +93,7 @@ void LoadMetadata(DkmProcess* process, WCHAR const* processPath, std::string_vie
 {
     auto winmd_path = path{ processPath };
     auto probe_file = std::string{ typeName };
-    do
+    for (;;)
```
Comment (sylveon) [inline natvis/cppwinrt_visualizer.cpp:96]: Idem - just use `while (true)` instead of do while tho

File: strings/base_implements.h [hunk 6]
```diff
@@ -692,7 +692,7 @@ namespace winrt::impl
         {
             uint32_t target = m_strong.load(std::memory_order_relaxed);
 
-            while (true)
+            for (;;)
```
Comment (sylveon) [inline strings/base_implements.h:695]: Idem

File: strings/base_implements.h [hunk 7]
```diff
@@ -924,7 +924,7 @@ namespace winrt::impl
             {
                 uintptr_t count_or_pointer = m_references.load(std::memory_order_relaxed);
 
-                while (true)
+                for (;;)
```
Comment (sylveon) [inline strings/base_implements.h:927]: Idem

File: strings/base_implements.h [hunk 8]
```diff
@@ -1048,7 +1048,7 @@ namespace winrt::impl
...
```
Comment (sylveon) [inline strings/base_implements.h:1051]: Idem

File: cppwinrt/cmd_reader.h (hunk 3 above)
Comment (kennykerr) [reply cppwinrt/cmd_reader.h:199]: Yes, I prefer `while (true)` but I assumed Clang warns about this. Is that correct?

Comment (sylveon) [reply cppwinrt/cmd_reader.h:199]: Doesn't seem so: https://godbolt.org/z/GE9cW8v3z

Comment (kennykerr) [reply cppwinrt/cmd_reader.h:199]: Sweet, then I'd prefer to keep the while loops.

**Comment by AtariDreams** [comment]
Fixed!

**Review by kennykerr — APPROVED** [review]
Thanks!

**Pull request merged**